src/pages/student/settings/page.tsx

```diff
@@ -1,11 +1,186 @@
-import { Layout } from "antd";
 import { PageHeader } from "../../../shared/ui/PageHeader/PageHeader";
+import { Button, Form, Input, Layout } from "antd";
+import { useUnit } from "effector-react";
+import {
+  toStudentPage,
+  setTelegramNickName,
+  setGithubNickName,
+  setRepositoryReact,
+  $telegramNickName,
+  $githubNickName,
+  $repositoryHtml,
+  $repositoryReact,
+  setRepositoryHtml,
+} from "../model";
+import {
+  repository,
+  Repository,
+} from './mocks';
+
+import "./page.css";
 
 const { Content } = Layout;
 
-export const StudentSettingsPage = () => (
+ 
+
+export const StudentSettingsPage = () => {
+  const [
+    telegramNickName,
+    githubNickName,
+    repositoryHtml,
+    repositoryReact,
+    toStudentPageFn,
+    setTelegramNickNameFn,
+    setGithubNickNameFn,
+    setRepositoryHtmlFn,
+    setRepositoryReactFn
+  ] = useUnit([
+    $telegramNickName,
+    $githubNickName,
+    $repositoryHtml,
+    $repositoryReact,
+    toStudentPage,
+    setTelegramNickName,
+    setGithubNickName,
+    setRepositoryHtml,
+    setRepositoryReact
+  ])
+  const handleFinish = () => toStudentPageFn();
+  const repositories = repository.map( obj => {
+    return (
+      <Form.Item
+
+        key={obj.description}
+        name={obj.name}
+        rules={[
+          {
+            required: true,
+            message: 'Please input name of repository!'
+          },
+          {
+            pattern: /[0-9a-z_]*$/,
+            message: "Please input name of repository!",
+          }
+        ]}
+        >
+    <Input
+      placeholder="имя репозитория с версткой"
+      value={obj.name}
+      onChange={({ target: { value } }) => setRepositoryFn(value)}
```

Comment:
If we use our store as an object, we will get rid of onChange and just use Form onFinish to sum up the settings